### PR TITLE
[OC-23] - created user service

### PR DESCRIPTION
**Pull request by eycwave**

### CODE REVIEW

File: src/main/java/com/quiz/model/User.java [hunk 1]
```diff
@@ -3,73 +3,48 @@
 import java.util.List;
 
 public class User {
-    private Long id;
-    private String fullName;
-    private String userName;
+    private int id;
+    private String fullname;
```
Comment (JamshidS) [inline src/main/java/com/quiz/model/User.java]: there is a standard in naming conversion in java called camel case

`userName`
`fullName`

File: src/main/java/com/quiz/repository/CreateTable.java [hunk 2]
```diff
@@ -84,10 +81,8 @@ public static void createQuizQuestion(){
         } catch (SQLException e) {
             e.printStackTrace();
         }
-
     }
 
-
     public static void createQuizTable(){
         try(Statement statement = DBConnectorConfig.getConnection().createStatement()){
             String query = "DROP SEQUENCE IF EXISTS quiz_id CASCADE;\n" +
```
Comment (JamshidS) [inline src/main/java/com/quiz/repository/CreateTable.java]: why you changed the sequence ?

File: src/main/java/com/quiz/service/UserService.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.quiz.service;
+
+import com.quiz.model.User;
+import com.quiz.model.UserQuiz;
+
+import java.util.List;
+
+public interface UserService {
+    User getUser(int userID);
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/UserService.java]: `getUserById(ing userID)`

File: src/main/java/com/quiz/service/UserService.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    List<User> getAllUser();
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/UserService.java]:
```suggestion
    List<User> getAllUsers();
```

File: src/main/java/com/quiz/service/impl/UserServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        List<User> allUser = userRepository.getAll();
+        if(allUser!=null){
+            return allUser;
+        }else{
+            throw new RuntimeException("User not found.");
+        }
+    }
+
+    public String updateUser(int userID, User updatedUser){
+        User user = userRepository.getById(userID);
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/UserServiceImpl.java:46]:
```java
List<User> users = userRepository.getAll();
```

File: src/main/java/com/quiz/service/impl/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        if(user!=null){
+            if(!user.getUsername().equals(updatedUser.getUsername())){
+                for(User testUsers : userRepository.getAll()){
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/UserServiceImpl.java]:
```suggestion
                for(User testUsers : users){
```

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/quiz/repository/CreateTable.java (hunk 2 above)
Comment (eycwave) [reply src/main/java/com/quiz/repository/CreateTable.java]: that wasn't mine this code was written by the friend before me. i changed it in "quiz service" branch.

File: src/main/java/com/quiz/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                isUserUnique(user);
+            }
+            userRepository.update(updatedUser, userID);
+            return "User successfully updated!";
+        }else{
+            throw new RuntimeException("User not found with ID: " + userID);
+        }
+    }
+
+    public void isUserUnique(User user){
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/UserServiceImpl.java]:
```suggestion
    public boolean isUserUnique(User user){
```

File: src/main/java/com/quiz/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+        if(allUser!=null){
+            return allUser;
+        }else{
+            throw new RuntimeException("User not found.");
+        }
+    }
+
+    public String updateUser(int userID, User updatedUser){
+        User user = userRepository.getById(userID);
+        List<User> users = userRepository.getAll();
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/UserServiceImpl.java]: remove this list from here if your not using it

**Review by JamshidS — CHANGES_REQUESTED** [review]

File: src/main/java/com/quiz/service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -44,7 +44,6 @@ public List<User> getAllUsers(){
 
     public String updateUser(int userID, User updatedUser){
         User user = userRepository.getById(userID);
-        List<User> users = userRepository.getAll();
         if(user!=null){
             if(!user.getUserName().equals(updatedUser.getUserName()) || !user.getEmail().equals(updatedUser.getEmail())){
                 isUserUnique(user);
```
Comment (JamshidS) [inline src/main/java/com/quiz/service/impl/UserServiceImpl.java]: it should be inside a condition

**Review by JamshidS — CHANGES_REQUESTED** [review]

**Review by JamshidS — APPROVED** [review]

**Pull request merged**